marshaling.go

```diff
@@ -132,6 +132,30 @@ type WorksheetConverter interface {
 var worksheetConverterType = reflect.TypeOf((*WorksheetConverter)(nil)).Elem()
 
 func (ws *Worksheet) StructScan(dest interface{}) error {
+	// mapping function returns ws field, ignore/skip, error
+	// this function uses tags
+	f := func(refWs *Worksheet, currentField reflect.StructField) (*Field, bool, error) {
+		tag, ok := currentField.Tag.Lookup("ws")
+
+		if !ok {
+			return nil, false, nil
+		}
+
+		if tag == "" {
+			return nil, false, fmt.Errorf("struct field %s: cannot have empty tag name", currentField.Name)
+		}
+
+		field, ok := refWs.def.fieldsByName[tag]
+		if !ok {
+			return nil, false, fmt.Errorf("unknown field %s", tag)
+		}
+
+		return field, true, nil
+	}
+	return ws.StructScanWithNameFunc(dest, f)
+}
+
+func (ws *Worksheet) StructScanWithNameFunc(dest interface{}, nameFn func(*Worksheet,reflect.StructField) (*Field, bool, error)) error {
```

Comment:
Can you provide a motivating example of why we'd want this API? It seems super specific, and in stark contrast from other Go APIs such as squalor, dat, json, yaml which use tag names.

Reply:
Motivation is to be able to define a deterministic rule for def field<->struct field naming, and not have to maintain tags (works well with autodefined defs).

The idea came from the reflect package which provides a similar function: FindByNameFunc.

Without this, I have to reimplement all of StructScan in our project, which is not a big deal, but the possibility of more control over name matching is what motivated me to implement worksheet refs and slices for StructScan, as I had similar functionality but wanted to be able to reuse the converter framework in here.

Reply:
The main StructScan function would remain tag-based, in line with standard marshaling in other frameworks.

One interest item is that other frameworks provide defaults based on the field name. That is, you never have to use the tag to make things work, there's a default mapping. That contrasts with marshaling here, where not having the tag means there is no scanning.

Reply:
yup, allowing uppercase + defaults + `ws:"-"` to ignore sounds like a perfect way to go